### DISCUSSION TITLE
How to get involved on the build automation + testing side of the project?

### DISCUSSION THREAD

**Discussion by nrvale0** [regolith-linux/regolith-desktop#962] [General]
Howdy. I'm not a developer per se but I have a long history doing various types of infrastructure automation and build+test automation. What's the best way to get involved with this project in those areas?

Transparently, this is selfishly motivated as I want to fix the problems I run into because I rely so heavily on regolith in my daily usage. 

thx

**Comment by kgilmer** [comment]
Hi @nrvale0 :)  At this time the project relies heavily on GitHub CI.  The top-level build/infra repo is here: https://github.com/regolith-linux/voulage/

There are some other distro targets w/ some prototype code but at this time Debian and Ubuntu packages are built from this workflow: https://github.com/regolith-linux/voulage/blob/main/.github/workflows/build-deb-v4.yml.  The high level summary of this would be:

1. for a given stage/distro/version/arch quad, calculate the package set to be build.  This package set is defined in trees of JSON, here: https://github.com/regolith-linux/voulage/tree/main/stage
2. after determining what packages should be built, compare against git hashs per repo from last build.  If no changes exit.
3. traverse over packages with updates and call `reprepro` to generate source and binary packages
4. rsync update back to package repo
5. generate release notes
6. run integration tests (at this time, only that some specified packages install cleanly into a fresh container of some target version)

The current areas I see in needing attention would be:
1. add more meaningful tests (eg screen capture after some scripted ui action to actuate login, etc.)
2. more distro support ~ build arch, gentoo, fedora, other Debian variants such as Pop and Mint
3. reliability, performance, robustness

**Reply by nrvale0** [reply]
I have been away from workstation for a few days. Just ACK'ing that I've seen this.

Digging in...
